pruner/find.go

```diff
@@ -31,8 +31,9 @@ func (s *Service) findPruneableHeaders(
 		return nil, err
 	}
 
-	if lastPruned.Height() == estimatedCutoffHeight {
-		// nothing left to prune
+	// estimatedCutoffHeight must be at least greater than lastPruned.Height()+1
+	// to continue, otherwise, there is nothing left to prune
+	if estimatedCutoffHeight <= lastPruned.Height()+1 {
```

Comment:
Why greater than lastPruned.Height() `+1` ? Seems like if it is lastPruned.Height()+1 it should be pruned and not skipped

Reply:
@walldiss let's say estimatedCutoffHeight == lastPruned.Height()+1 where lastPruned height is 5


The request down to the header store would look like: 

GetRangeByHeight(from:5, 6) -- it goes down to the header store as: 



```
func (s *Store[H]) GetRangeByHeight(
	ctx context.Context,
	from H,
	to uint64,
) ([]H, error) {
	headers, err := s.getRangeByHeight(ctx, from.Height()+1, to)
	if err != nil {
		return nil, err
	}
```

`getRangeByHeight( 6, 6)` where it hits



```
	if from > to-1 {
		return nil, fmt.Errorf("header/store: invalid range(%d,%d)", from, to-1)
	}
```

if 6 > 5 



It's just ugliness from inconsistent implementations across GetRangeByHeight implementations and what range exactly is expected as a return value.

Reply:
Request has to be from 6, as it is the height that needs to be pruned. GetRangeByHeight(from:6, 6)

Reply:
I think the issue is in header pkg. It should include `from` into requested interval

Reply:
It shouldn't because otherwise you request data you already have